terraform/cp4ba/variables.tf

```diff
@@ -0,0 +1,232 @@
+}
+
+variable "public_vlan_number" {
+  default     = ""
+  description = "**Classic Only**. Ignored if `cluster_id` is specified. Public VLAN assigned to your zone. List available VLANs in the zone: `ibmcloud ks vlan ls --zone <zone>`, make sure the the VLAN type is public and the router begins with fc. Use the ID or Number. Leave blank if Public VLAN does not exist, one will be created"
+}
+
+variable "storage_capacity"{
+    type = number
+    default = 200
+    description = "Ignored if Portworx is not enabled: Storage capacity in GBs"
+}
+
+variable "storage_db2" {
+    type = number
+    default = 10
+    description = "Ignored if Portworx is not enabled. Optional, Used only if a user provides a custom storage_profile"
+}
+
+variable "storage_profile" {
+    type = string
+    default = "10iops-tier"
+    description = "Ignored if Portworx is not enabled. Optional, Storage profile used for creating storage"
+}
+
+variable "create_external_etcd" {
+    type = bool
+    default = false
+    description = "Ignored if Portworx is not enabled: Do you want to create an external etcd database? `true` or `false`"
+}
+
+# These credentials have been hard-coded because the 'Databases for etcd' service instance is not configured to have a publicly accessible endpoint by default.
+# You may override these for additional security.
+variable "etcd_username" {
+  default = ""
+  description = "Ignored if Portworx is not enabled: This has been hard-coded because the 'Databases for etcd' service instance is not configured to have a publicly accessible endpoint by default.  Override these for additional security."
+}
+
+variable "etcd_password" {
+  default = ""
+  description = "Ignored if Portworx is not enabled: This has been hard-coded because the 'Databases for etcd' service instance is not configured to have a publicly accessible endpoint by default.  Override these for additional security."
+}
+
+variable "cluster_config_path" {
+  default     = "./.kube/config"
+  description = "directory to store the kubeconfig file"
+}
+
+
+//variable "cluster_id" {
+//  default     = ""
+//  description = "Enter your cluster id or name to install the Cloud Pak. Leave blank to provision a new Openshift cluster."
+//}
+
+// Portworx Module Variables
+variable "install_portworx" {
+  type        = bool
+  default     = false
+  description = "Install Portworx on the ROKS cluster. `true` or `false`"
+}
+
+locals {
+  cp4ba_namespace              = "cp4ba-project"
+  entitled_registry            = "cp.icr.io"
+  entitled_registry_user       = "cp"
+  enable_cluster               = var.cluster_name_id == "" || var.cluster_name_id == null
+  use_entitlement              = var.use_entitlement #? ["yes", "Yes", "YES", "y", "Y"] : ["", "n", "N", "no", "No", "NO"]
```

Comment:
Why are `use_entitlement`, `local_public_registry_server`, `local_registry_server`, `local_registry_user`,  `local_registry_password` local variables?  If you aren't doing anything different to them, just refer to the var.* in the module call.